### PR TITLE
MAP-896 Add deactivation details and update location indexes


### PR DESCRIPTION
**Pull request by Mjwillis**
Added deactivation details such as deactivatedDate, deactivationReason, and proposedReactivationDate to SyncAndMigrateResourceIntTest and UpsertLocationRequest files. Also created new indexes and dropped the old index in the V1_13__location_indexes.sql file.

In addition to updating the deactivated attribute from active to false for a location, deactivation details are now added to ensure better tracking and management of deactivated locations. Modifying the database indexes is intended to improve query performance related to location and location history.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/dto/UpdateLocationRequest.kt [hunk 1]
```diff
@@ -20,4 +20,5 @@ interface UpdateLocationRequest {
   val deactivationReason: DeactivatedReason?
   val proposedReactivationDate: LocalDate?
   val deactivatedDate: LocalDate?
+  fun isDeactivated() = deactivationReason != null && deactivatedDate != null
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/dto/UpdateLocationRequest.kt:23]: hmm yes this was worrying me a bit, Ive just checked and there are a lot of rows in Nomis where the 3 deactivated/ reactivate columns are null but active is false too. Might need an explicit 'active' field?

Comment (steverendell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/dto/UpdateLocationRequest.kt:23]: Or .. I could set the reason as 'OTHER' in these cases?

Comment (Mjwillis) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/dto/UpdateLocationRequest.kt:23]: Ah ok, or if not active have a deactivation of OTHER with a last modified date for deactivation?

**Review by steverendell — APPROVED** [review]
lgtm

**Pull request merged**